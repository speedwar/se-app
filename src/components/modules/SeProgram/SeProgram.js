import React, { useContext}  from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { SeCard } from 'components/components'
import { ThemeContext } from 'context'

import imgTile from 'assets/images/tile.jpg'
import imgLogo from 'assets/images/logo.png'

const SeProgram = ({ title }) => {
  const themeContext = useContext(ThemeContext)

  const mockData = {
    data: {
      programs: [
        {
          id: 0,
          title: 'Home and Away',
          provider: 'Seven',
          image: imgTile,
          logo: imgLogo,
        }
      ]
    }
  }

  // Duplicating a program multiple times for DEMO
  let programList = [...mockData.data.programs]
  let loopNumber = 6;
  while(--loopNumber) {
    programList.push(...mockData.data.programs)
  }

  const SeProgramTitle = styled.h2`
    color: ${themeContext.theme.main.text};
    font-size: 1.25rem;
  `
  return (
    <SeProgramContainer>
      <SeProgramTitle>
        { title }
      </SeProgramTitle>
      <SeProgramContent>
        {programList.map((program, i) => (
          <SeCard
            key={ program.title + i }
            title={ program.title }
            provider= { program.provider }
            imgTile={ program.image }
            imgLogo={ program.logo }
          />
        ))}
      </SeProgramContent>
    </SeProgramContainer>
  )
}

SeProgram.propTypes = {
  imgLogo: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
  ]),
  imgTile: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
  ]),
  title: PropTypes.string.isRequired,
  provider: PropTypes.string,
}

export default SeProgram

const SeProgramContainer = styled.section``

const SeProgramContent = styled.div`
  margin-right: -0.5em;
  margin-left: -0.5em;
`
