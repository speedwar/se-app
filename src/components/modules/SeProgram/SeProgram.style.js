import styled from 'styled-components'

export const SeProgramContainer = styled.section``

export const SeProgramContent = styled.div`
  margin-right: -0.5em;
  margin-left: -0.5em;
`
