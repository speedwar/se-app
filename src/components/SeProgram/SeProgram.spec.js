import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import SeProgram from './SeProgram'

const defaultProps = {
  title: 'Programs',
}

const setup = (partialProps) => {
  const props = {
    ...defaultProps,
    ...partialProps,
  }
  const wrapper = shallow(<SeProgram { ... props } />)
  return { wrapper, props }
}

describe('<SeProgram /> component', () => {
  it('renders component with props', () => {
    const wrapper = setup({
      title: 'Now On'
    })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.props.title).toEqual('Now On')
  })
})
