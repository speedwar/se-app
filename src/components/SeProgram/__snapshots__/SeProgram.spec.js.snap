// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeProgram /> component renders component with props 1`] = `
Object {
  "props": Object {
    "title": "Now On",
  },
  "wrapper": <SeProgram__SeProgramContainer>
    <SeProgram__SeProgramTitle>
      Now On
    </SeProgram__SeProgramTitle>
    <SeProgram__SeProgramContent>
      <SeCard
        href="#"
        imgLogo={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        imgTile={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        key="Home and Away0"
        provider="Seven"
        title="Home and Away"
      />
      <SeCard
        href="#"
        imgLogo={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        imgTile={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        key="Home and Away1"
        provider="Seven"
        title="Home and Away"
      />
      <SeCard
        href="#"
        imgLogo={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        imgTile={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        key="Home and Away2"
        provider="Seven"
        title="Home and Away"
      />
      <SeCard
        href="#"
        imgLogo={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        imgTile={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        key="Home and Away3"
        provider="Seven"
        title="Home and Away"
      />
      <SeCard
        href="#"
        imgLogo={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        imgTile={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        key="Home and Away4"
        provider="Seven"
        title="Home and Away"
      />
      <SeCard
        href="#"
        imgLogo={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        imgTile={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
        key="Home and Away5"
        provider="Seven"
        title="Home and Away"
      />
    </SeProgram__SeProgramContent>
  </SeProgram__SeProgramContainer>,
}
`;
